### PR TITLE
[docs] Add a demo about how to debounde `onChange`

### PR DESCRIPTION
**Pull request by alexfauquette**
Part of #8375

### CODE REVIEW

File: docs/data/date-pickers/base-concepts/ServerInteraction.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        setValue(newValue);
+        deboucedOnAccept(newValue);
+        onChange?.(newValue, context);
+      }}
+      {...other}
+    />
+  );
+}
+
+const adapter = new AdapterDayjs();
```
Comment (flaviendelangle) [inline docs/data/date-pickers/base-concepts/ServerInteraction.tsx:110]: Can we move this definition at the top of the file ? 
And I would be in favor of not using the adapter at all but to use directly the date library API, which is what we usually do.

File: docs/data/date-pickers/base-concepts/ServerInteraction.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+export default function ServerInteraction() {
+  const [onChangeLogs, setOnChangeLogs] = React.useState<(Dayjs | null)[]>([]);
```
Comment (flaviendelangle) [inline docs/data/date-pickers/base-concepts/ServerInteraction.tsx:112]: An idea to avoid having `on` as the prefix of a state, which is a little weird

```suggestion
  const [logsFromOnChange, setLogsFromOnChange] = React.useState<(Dayjs | null)[]>([]);
```

**Comment by flaviendelangle** [comment]
Would it make sense to move this demo in #8372 ?

**Comment by alexfauquette** [comment]
> Would it make sense to move this demo in the new page of https://github.com/mui/mui-x/pull/8372 ?

Yes, would you like that I directly move those commits to the #8372 PR?

**Comment by flaviendelangle** [comment]
Yes why not :+1: 
I can move it to avoid GIt conflicts

File: docs/data/date-pickers/base-concepts/base-concepts.md [hunk 3]
```diff
@@ -96,6 +96,25 @@ A correct `aria-labelledby` value is assigned to the dialog component based on t
...
 
+## Server interaction
+
+If the selected value is used to interact with server, you might want to avoid sending all the intermediate states.
+Especially if the user is setting the date thanks to arrow interaction.
+
+In such a case, the recommended UI is to add a button for validating the form.
+If for some reason, you need to send the data to the server without having the user pressing a validation button, you can debouce the `onChange` as follow.
+
+In the next demos, `<DateField />` is extended to add an `onAccept` props which is a debounced version of the `onChange`.
```
Comment (LukasTy) [inline docs/data/date-pickers/base-concepts/base-concepts.md:107]:
```suggestion
In the following demo, `<DateField />` is extended adding an `onAccept` prop which is a debounced version of the `onChange`.
```

File: docs/data/date-pickers/base-concepts/ServerInteraction.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  logsFromOnAccept: (Dayjs | null)[];
+};
+function DisplayEvents(props: DisplayEventsProps) {
+  const { logsFromOnChange, logsFromOnAccept } = props;
+  return (
+    <Stack direction="row" justifyContent="space-between" sx={{ p: 2 }}>
+      <Stack sx={{ minWidth: 180 }}>
+        <Typography>onChange</Typography>
+        <Divider />
+        <Typography component="pre" sx={{ maxHeight: '15rem', overflow: 'auto' }}>
```
Comment (LukasTy) [inline docs/data/date-pickers/base-concepts/ServerInteraction.tsx:23]: Have you considered extracting this layout into another component as well?
IMHO it would make sense and at the same time—it would avoid the issue of `onAccept` section not having maxHeight and increasing the container height after enough delayed field modifications 🙈 
Screenshot 2023-04-17 at 13 16 06

File: docs/data/date-pickers/base-concepts/ServerInteraction.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+            )
+            .join('\n')}
+        </Typography>
+      </Stack>
+    </Stack>
+  );
+}
+
+// debounce function
+function debounce(func: (...arg: any) => void, wait = 500) {
```
Comment (LukasTy) [inline docs/data/date-pickers/base-concepts/ServerInteraction.tsx:65]: Have you considered using `debounce` from `@mui/utils` instead? 🤔

File: docs/data/date-pickers/base-concepts/ServerInteraction.tsx [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <Stack sx={{ textAlign: 'center', alignItems: 'center' }}>
+        <DateFieldWithAccept
+          onAccept={onAccept}
+          onChange={(newValue) => {
+            setLogsFromOnChange((prev) => [newValue, ...prev]);
+          }}
+          sx={{ width: 150 }}
+          label="debounced field"
+          defaultValue={dayjs(new Date(2020, 10, 5))}
```
Comment (LukasTy) [inline docs/data/date-pickers/base-concepts/ServerInteraction.tsx:133]: Have you considered using the same default as in most other demos? 🤔 
```suggestion
          defaultValue={dayjs('2022-04-17')}
```

**Review by LukasTy — COMMENTED** [review]
Great addition! 🙌

File: docs/data/date-pickers/base-concepts/ServerInteraction.tsx (hunk 5 above)
Comment (alexfauquette) [reply docs/data/date-pickers/base-concepts/ServerInteraction.tsx:65]: It's a copy paste of `@mui/utils` code. But I preferred to show the code than importing a function which is not documented

Comment (flaviendelangle) [reply docs/data/date-pickers/base-concepts/ServerInteraction.tsx:65]: If the debounce is a point of detail of the demo, I'm in favor of importing from `@mui/utils`

But here, it's the main point of the demo so I think re-defining it is a good idea.

File: docs/data/date-pickers/base-concepts/base-concepts.md (hunk 3 above)
Comment (alexfauquette) [reply docs/data/date-pickers/base-concepts/base-concepts.md:107]: @samuelsycamore could you give some hints on this topic?

I noticed that each time I want to express "the thing X has been done to get the result Y" I end up with the structure "X is <verb+ED> to <verb> Y" which Grammarly frequently corrects by "X is <verb+ED> <verb+ING> Y"

My problem is that: 
the first sentence feels like "X is done in order to get Y" (what I want to express)
the corrected sentence feels like "X is done the consequence being Y"

I don't know the correct expression, and if there is a better way to express this kind of idea.

Comment (samuelsycamore) [reply docs/data/date-pickers/base-concepts/base-concepts.md:107]: Grammarly can be helpful sometimes but it's not always trustworthy. It has odd idiosyncrasies and misunderstands what's being communicated. In this case the structure doesn't make sense. I think it stems from a larger problem, which is that the sentence itself uses passive voice, and Grammarly doesn't understand how the clauses are related. 

Here's how I would rephrase it to use active voice instead:

```suggestion
The following demo shows how to extend the Date Field component by adding an `onAccept` prop, which is a debounced version of `onChange`.
```
